[@mantine/dates] fix timeInputProps defaultValue not working

Cuzart opened the pull request:
Fixes #6958 and is self-explanatory

rtivital commented:
This won't work, if the default value is set on the TimeInput and then not modified by the user, it will not be treated as part of the date object passed to `onChange` handler.

Cuzart commented:
Yes you are right I missed that point. Out of interest I tested an approach in the handleDateChange function if we check if the formattedTime is different from the timeInputProps?.defaultValue we get the expected behaviour. But this could add a bit of complexity depending on the format of the defaultValue and as you mentioned you are not planing to support this any time soon I will try to fix this outside of the component via JavaScript. Anyhow it could be misleading that the type of timeInputProps is supporting defaultValue then.

```tsx
const handleDateChange = (date: DateValue) => {
    if (date) {
      if (timeInputProps?.defaultValue && formatTime(date) !== timeInputProps?.defaultValue) {
        date.setHours(parseInt(timeInputProps?.defaultValue?.split(':')[0]));
        date.setMinutes(parseInt(timeInputProps?.defaultValue?.split(':')[1]));
      }
      setValue(assignTime(_value, date));
    }
    timeInputRef.current?.focus();
  };

```

rtivital commented:
Okay, I'll remove defaultValue and value props from this type
